primitives/teerex/src/lib.rs

```diff
@@ -111,6 +111,16 @@ impl QuotingEnclave {
 			tcb,
 		}
 	}
+
+	pub fn attributes_flags_mask_as_u64(&self) -> u64 {
+		let slice_as_array: [u8; 8] = self.attributes_mask[0..8].try_into().unwrap();
```

Comment:
I had the same thing in my mind because of recent PRs as well :D.

I would say yes, because the `QuotingEnclave` look like this:


```rust
pub struct QuotingEnclave {
// FIELDS OMITTED HERE
	pub miscselect: [u8; 4],
	pub miscselect_mask: [u8; 4],
        pub attributes: [u8; 16],
	pub attributes_mask: [u8; 16],
// FIELDS OMITTED HERE
}
```
An instantiated `QuotingEnclave` is guaranteed to have `attributes` and `attributes_mask` fields with a size of 16, or else the compiler will complain `expected an array with a fixed size of X elements, found one with Y elements`.

Reply:
Hahaha, yes you are right. This is fine then. 👍